### PR TITLE
add proxy support

### PR DESCRIPTION
**Pull request by parrasajad**
### Examples
socks5:
```console:
$ echo AS54115 | asnmap --proxy socks5://127.0.0.1:40932
```
http:
```console:
$ echo AS54115 | asnmap --proxy http://127.0.0.1:8080
```
input file support:
```console:
$ echo AS54115 | asnmap --proxy proxtlist.txt
```

### CODE REVIEW

File: libs/client.go [hunk 1]
```diff
@@ -68,6 +75,73 @@ func NewClient() *Client {
 	return &client
 }
 
+// SetProxy adds a proxy to the client
+func (c *Client) SetProxy(proxyList []string) {
```
Comment (Mzack9999) [inline libs/client.go:79]: The signature should return an error if no valid proxy was found

File: libs/client.go [hunk 2]
```diff
@@ -68,6 +75,73 @@ func NewClient() *Client {
...
+list:
+	for _, p := range proxyList {
+		proxyurl, err := url.Parse(p)
```
Comment (Mzack9999) [inline libs/client.go:82]: I think this code can be moved into a proper function since it's repeated at line 103

File: libs/client.go [hunk 3]
```diff
@@ -68,6 +75,73 @@ func NewClient() *Client {
...
+					continue
+				}
+				defer file.Close()
+				scanner := bufio.NewScanner(file)
+				for scanner.Scan() {
+					proxy := scanner.Text()
+					if strings.TrimSpace(proxy) == "" {
+						continue
+					}
+					proxyurl, err := url.Parse(proxy)
```
Comment (Mzack9999) [inline libs/client.go:103]: I think this code can be moved into a proper function since it's repeated at line 82

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Please try to refactor the `SetProxy` method by reducing complexity and increase code readability, for example:
- Move URL checking and setproxy for each item into a proper routine
- Move the file process utility into a subroutine
- Reduce `if` nesting by preferring `err != nil` checks and early return

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]

**Review by Gitgody — COMMENTED** [review]
Exit

**Comment by Gitgody** [comment]
> ### Examples
> 
> socks5:
> 
> ```console:
> 
> $ echo AS54115 | asnmap --proxy socks5://127.0.0.1:40932
> 
> ```
> 
> http:
> 
> ```console:
> 
> $ echo AS54115 | asnmap --proxy http://127.0.0.1:8080
> 
> ```
> 
> input file support:
> 
> ```console:
> 
> $ echo AS54115 | asnmap --proxy proxtlist.txt
> 
> ```

Exit

**Pull request merged**